packages/eslint-plugin/tests/rules/no-useless-template-literals.test.ts

```diff
@@ -0,0 +1,303 @@
+import { noFormat, RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-useless-template-literals';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-useless-template-literals', rule, {
+  valid: [
+    "const string = 'a';",
+
+    'const string = `a`;',
+
+    `
+      const string = 'a';
+      const concatenated = \`\${string}b\`;
```

Comment:
[Style] This is non-blocking, but just in case you prefer any of these:

* Instead of declaring variables, you can use `declare`. Example: `declare const string: 'a';`. It's a bit more semantically close to what the test is trying to do (use an identifier with a particular type). Though admittedly it's more characters...
   * (I do see there are some cases that do this now - reposting this point so it's alongside the others)
* Omit the variable for a template literal, and just have it as the statement. Example: <code>\`\\${string}b\`;</code>.
* If there's only one line in the test case, not have an extra blank line in front & end.

I won't request changes on any of these. Just figured I'd mention them 😄

Reply:
1. declare is longer, yeah, but I don't really have a preference about it, so why not
2. hmm, I gotta admit that having a statement without a variable seems weird to me, but same as above, don't really have any strong preference
3. I've changed some of the cases but kept others with empty lines. I think sometimes it's not readable and feels too condensed.

See this: 
![image](https://github.com/typescript-eslint/typescript-eslint/assets/32631382/8b0f4b9a-0b9e-430c-bd02-2d150ae02863)

Versus this:
![image](https://github.com/typescript-eslint/typescript-eslint/assets/32631382/706c71a7-116d-44cf-a1d5-d0f5b4b7fd6f)